packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2460,6 +2480,9 @@ class InputDecoration {
   }) : assert(enabled != null),
        assert(!(prefix != null && prefixText != null), 'Declaring both prefix and prefixText is not supported.'),
        assert(!(suffix != null && suffixText != null), 'Declaring both suffix and suffixText is not supported.'),
+       // ignore: deprecated_member_use_from_same_package
+       assert(!(!hasFloatingPlaceholder && identical(floatLabelBehavior, FloatLabelBehavior.always)),
```

Comment:
Please use `==` instead of identical here and below

Reply:
Using `identical` is required here due to this issue: https://github.com/dart-lang/sdk/issues/29278

Reply:
I didn't realize that, thanks for the pointer!